profiles/common/modules/custom/ecas/includes/ecas.inc

```diff
@@ -514,13 +531,27 @@ function ecas_login_check() {
         ecas_login_page();
       }
     }
+    // If we have a good user.
+    else {
+      unset($_SESSION['phpCAS']);
+      $user = drupal_anonymous_user();
+    }
   }
-  // If we have a good user.
-  else {
-    unset($_SESSION['phpCAS']);
-    $user = drupal_anonymous_user();
-  }
+}
 
+/**
+ * Internal function to unset destination from request.
+ */
+function _ecas_unset_destination() {
+  // We cannot ensure that the user is the same as in Ecas,
+  // So we absolutely want to redirect the user
+  // to the ECAS warning page - thus, we unset any previously set destination.
+  if (isset($_REQUEST['destination'])) {
```

Comment:
Function drupal_goto() works with $_GET['destination'] .
I'm not sure if changing $_REQUEST['destination'] you have the desired effect.

Reply:
This has been removed